Mixpanel/AutomaticEvents.swift

```diff
@@ -86,9 +86,11 @@ class AutomaticEvents: NSObject, SKPaymentTransactionObserver, SKProductsRequest
                                                object: nil)
 
         SKPaymentQueue.default().add(self)
-        DispatchQueue.main.async {
-            if self.automaticPushTracking {
-                self.setupAutomaticPushTracking()
+        DispatchQueue.main.async { [weak self] in
+            if let hasSelf = self {
```

Comment:
In swift 4.2,  using optional binding to upgrade self from a weak to strong reference is supported, so you can just 
```suggestion
            if let self = self {
```
This is preferred by us to avoid inconsistent naming for strong self.
https://github.com/apple/swift-evolution/blob/master/proposals/0079-upgrade-self-from-weak-to-strong.md#proposed-solution

Reply:
Could you help to replace all occurrences?

Reply:
Hmmm, I was under the impression that aliasing let self = self we as a compiler bug, and using a different name for the local strong self was the preferred syntax. I'm happy to change it, but this seems likely to break with future versions of swift. 

https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20160118/007425.html

I will change it back in just a sec